### ISSUE TITLE
Cannot Add New Inline Tags Due to Prohibitive Slowness ( > 15 sec. lag)

### ISSUE THREAD

**Issue by brandon-carag** [glushchenko/fsnotes#957]
**Describe the bug**
Adding a tag to a new note is prohibitively slow (takes approximately 15 seconds for one new tag).

**To Reproduce**
1. Create a note library of arbitrary text (approximately 2k notes in my case) and tag each note with a multi-level label like so referencing a random date `#created_at/2019/05/02`
2. Ensure inline tag scanning is enabled in preferences
3. Create a new note and try to add a new tag directly in the note text via the keyboard (eg NOT via the tag GUI).  

**Expected behavior**
New tag additions should be instantaneous

**Actual behavior**
A spinning beach ball occurs while adding a new tag and FS Notes freezes for about 15 seconds (presumably while the lookup completes), making it prohibitively slow to add new tags.

**Workaround**
Turn off inline tag scanning

**Supporting Files**
None

**Desktop (please complete the following information):**
 - OS: macOS
 - OS version: 10.14.6
 - FSNotes version: 4.5.1

**Additional context**
The largest note in my note library is about 37 KB.  Approximately 80% of my notes are under 3 KB each.

**Comment by glushchenko** [comment]
Can not reproduce this.

**Comment by brandon-carag** [comment]
Thanks for trying to replicate.  Great product btw, I really like FS Notes (I'm a convert from Nvalt), just which it was a bit more stable.

I just copied my note library to another machine and have observed the exact same behavior there.  I also tried taking iCloud out of the equation in case syncing was an effect by changing the Notes directory to a totally separate local directory.

I suppose I could script something to delete all my notes out except for the tags and could attach the note library here.  Would that be helpful?

**Comment by glushchenko** [comment]
@brandon-carag, think this will help solve the problem. I'll be very thankful.

**Comment by brandon-carag** [comment]
No problem.  Good news--I wrote a script to simplify/sanitize my personal notes and was able to achieve replication.  This should help isolate the issue for you since there are no longer any code blocks, things that might be proprietary to my notes, images, etc.

This is the sanitized format of the notes:
- Approximately 2000ish notes
- They all have random UUIDs as the note title
- I've retained a `fs_at/2018/12/14` style tag in each note
- I've added a filler line of arbitrary text where the standard note text would be.  I wanted to emulate the size of my actual note library as close as possible.
- Total size of the files is 12 MB

I'll be sending the a zip with the note contents to fluder@icloud.com.  I think my script sanitized all my notes, but since they are based on my personal notes library I didn't want to accidentally miss something and leave it publicly on github.

Let me know if you have any questions.  Thanks again for your speedy feedback here.

**Comment by brandon-carag** [comment]
I got a message back from fluder@icloud.com that the message was truncated since there's no longer space for file attachments, so here's an encrypted zip file.  I send the password for decrypting via email referencing this github issue.

[encrypted_sanitized_notes.zip](https://github.com/glushchenko/fsnotes/files/4992794/encrypted_sanitized_notes.zip)

Thanks!

**Comment by glushchenko** [comment]
@brandon-carag, please try this:

https://github.com/glushchenko/fsnotes/releases/download/4.5.3/FSNotes_4.5.3.zip

**Comment by brandon-carag** [comment]
Hey there--that's an order of magnitude faster when I tested on my notes library.  It's not quite instantaneous, but definitely MUCH faster (takes perhaps half a second now instead of 15s +).  Thank you!

On a related note, I've noticed that when collapsing and uncollapsing nested tags in the sidebar, the UI freezes up--for example, if you have 2015/12/01 uncollapsed, toggling "01" doesn't refresh on screen.  I'm happy to open another ticket if helpful (I don't want to muddy things), but wanted to mention it since you already have a copy of the Notes Library that's showing those issues.  Thank you again!
